### ISSUE TITLE
readthedocs pdf build is failing

### ISSUE THREAD

**Issue by hvy** [optuna/optuna#1795]
The pdf builds on readthedocs have been failing for the last couple of days. It was successfully built for v2.0.0 but is broken in `master` as well as in the latest v2.1.0 release. 

It might be worth trying to reproduce the error with the readthedocs' official Docker image https://docs.readthedocs.io/en/stable/builds.html#docker-images.

**Comment by harupy** [comment]
[The failed build log](https://readthedocs.org/projects/optuna/builds/11765639) says:

```
! pdfTeX error (ext4): \pdfendlink ended up in different nesting level than \pd
fstartlink.
\AtBegShi@Output ...ipout \box \AtBeginShipoutBox 
                                                  \fi \fi 
l.4203 \subsubsection*{Methods}
                               
!  ==> Fatal error occurred, no output PDF file produced!
```

[This page](https://www.overleaf.com/learn/latex/Questions/What_does_%22%5Cpdfendlink_ended_up_in_different_nesting_level_than_%5Cpdfstartlink%22_mean%3F) says the error message means:

> The error message "\pdfendlink ended up on a different page" is usually due to a hyperlink (perhaps a citation, or a URL, or cross-reference) somewhere in your PDF, **that's breaking across two page (i.e. starting near the end of one page and end on the next). (Citations are possible culprits.) Unfortunately, pdfLaTeX can't deal with this**.

**Comment by harupy** [comment]
The doc build has been failing since this commit: https://github.com/optuna/optuna/commit/05365bbf17dcff0bc0bf242ff23cc3b77677b932

**Comment by hvy** [comment]
Thanks for your quick debugging. Do you know how we can address the issue, for instance if there's some configuration in sphinx that we can utilize? My understanding is that it could start to occur when we make a change to the documentation anywhere, and we additionally can't tell exactly where the error occurs.

**Comment by harupy** [comment]
I have no idea why this error occurs lol. One possible reason is that the example added by https://github.com/optuna/optuna/commit/05365bbf17dcff0bc0bf242ff23cc3b77677b932 pushes donw the hyperlinks in `Args`, `Returns` and `See also` sections and one of them breaks across two pages.

**Comment by toshihikoyanase** [comment]
The error mentioned in https://github.com/optuna/optuna/issues/1795#issuecomment-688930016 has been resolved since Sept. 17. Please refer to [this log](https://readthedocs.org/projects/optuna/builds/11901523/). Just FYI, the last failure is [this build](https://readthedocs.org/projects/optuna/builds/11765639/).

As we can see the return code of `pdflatex`, the PDF build has still some minor issues, but we can successfully build the PDF now.

```console
Collected error summary (may duplicate other messages):
  pdflatex: Command for 'pdflatex' gave return code 256
```

The actual errors are as follows:

*1. Sphinx’s pdflatex does not support svg graphics*

```
! LaTeX Error: Unknown graphics extension: .svg.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...

l.171 ...c99b38ea299ee813f5e736/colab-badge}.svg}}
```

*2. Syntax error of math expression*

```
! Undefined control sequence.
l.2213 \lt
           \mathsf{high}\), where \(k\) is an integer.
The control sequence at the end of the top line
of your error message was never \def'ed. If you have
misspelled it (e.g., `\hobx'), type `I' and the correct
spelling (e.g., `I\hbox'). Otherwise just continue,
and I'll forget about whatever was undefined.
```


*3. Illegal character*

```
! Package inputenc Error: Unicode char 　 (U+3000)
(inputenc)                not set up for use with LaTeX.

See the inputenc package documentation for explanation.
Type  H <return>  for immediate help.
 ...
```

Here's a reproducible code:

```console
cd /path/to/optuna
docker run -it -v $(pwd):/prj --rm -w /prj readthedocs/build:latest bash
pip3 install -e ."[document]"
cd docs/source/
sphinx-build -b latex -D language=en -d _build/doctrees . _build/latex
cd /prj/docs/source/_build/latex
latexmk -r latexmkrc -pdf -f -dvi- -ps- -jobname=optuna -interaction=nonstopmode
```

**Comment by HideakiImamura** [comment]
The errors 2 and 3 pointed by @toshihikoyanase have addressed by @KoyamaSohei in #2067 and #2068. Thank you!

Error 1 is waiting to be addressed.

**Comment by toshihikoyanase** [comment]
@hvy The errors in https://github.com/optuna/optuna/issues/1795#issuecomment-728605927 were resolved by #2067, #2068 and #2323. Can we close this issue?

**Comment by hvy** [comment]
Thanks for the heads up, yes I think we can close it.
